pkgs/applications/networking/instant-messengers/viber/unwrapped.nix

```diff
@@ -0,0 +1,155 @@
+{ lib
+, fetchurl
+, makeBinaryWrapper
+, autoPatchelfHook
+, dpkg
+, alsa-lib
+, nss
+, gst_all_1
+, stdenvNoCC
+, xorg
+, lcms
+, libwebp
+, snappy
+, gdk-pixbuf
+, libmng
+, gtk3
+, pango
+, libtiff
+, tslib
+, mtdev
+, libxslt
+, libpulseaudio
+, mesa
+, xcb-util-cursor
+, libxkbcommon
+, libglvnd
+, brotli
+, cups
+, libevent
+, fontconfig
+, freetype
+, libkrb5
+, harfbuzz
+, nspr
+, python311Packages
+, zstd
+, udev
+}:
+
+stdenvNoCC.mkDerivation {
+
+  pname = "viber";
+  version = "21.0.0.1";
+  meta = {
+    homepage = "https://www.viber.com";
+    description = "An instant messaging and Voice over IP (VoIP) app";
+    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
+    license = lib.licenses.unfree;
+    platforms = [ "x86_64-linux" ];
+    maintainers = with lib.maintainers; [ jagajaga liarokapisv ];
+  };
+
+  src = fetchurl {
+    # Official link: https://download.cdn.viber.com/cdn/desktop/Linux/viber.deb
+    url = "https://web.archive.org/web/20231030150355/https://download.cdn.viber.com/cdn/desktop/Linux/viber.deb";
+    sha256 = "sha256-47E6emeIYyG0jid7Oqrn2CEKtb9nQhOckFtEMBevZGA=";
+  };
+
+  # Needed due to failing dlopen calls.
+  appendRunpaths = [
+    "${udev}/lib"
+  ];
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    dpkg
+    makeBinaryWrapper
+  ];
+
+  buildInputs = [
+    alsa-lib
+    brotli.lib
+    cups.lib
+    libevent
+    fontconfig.lib
+    freetype
+    gtk3
+    gdk-pixbuf
+    libkrb5
+    gst_all_1.gst-plugins-bad
+    gst_all_1.gst-plugins-base
+    gst_all_1.gstreamer
+    gtk3
+    harfbuzz
+    xorg.libICE
+    lcms
+    libmng
+    mtdev
+    nspr
+    nss
+    pango
+    xorg.libSM
+    snappy
+    libtiff
+    libwebp
+    xorg.xcbutilwm
+    xorg.xcbutilkeysyms
+    xorg.libxkbfile
+    xorg.libXrandr
+    libxslt
+    python311Packages.libxslt
+    xorg.libXScrnSaver
+    xorg.libXtst
+    zstd
+    libxkbcommon
+    libglvnd
+    libpulseaudio
+    mesa
+    xcb-util-cursor
+    tslib
+  ];
+
+  installPhase =
+    let
+      gstreamerPluginPath = lib.makeSearchPath "lib/gstreamer-1.0/" [
+        gst_all_1.gstreamer.out
+        gst_all_1.gst-plugins-base
+        gst_all_1.gst-plugins-good
+        gst_all_1.gst-plugins-ugly
+        gst_all_1.gst-plugins-bad
+      ];
+    in
+    ''
+      dpkg-deb -x $src $out
+      mkdir -p $out/bin
+
+      # Soothe nix-build "suspicions"
+      chmod -R g-w $out
+
+      # GST_PLUGIN_SYSTEM_PATH is specified because Qt6's QGStreamerMediaPlayer calls
+      # gst_element_factory_make which in turn uses the env variable to perform the lookup.
+      # Qt6 uses the returned object without checking for NULL which will lead to a SEGFAULT.
+
+      wrapProgram $out/opt/viber/Viber \
```

Comment:
Isn't this contained in wrapQt or similar?

Reply:
No. This is an issue with [QtMultimedia](https://github.com/qt/qtmultimedia/blob/697a12f91166b3985756113bc9f88825f2ecc9ea/src/plugins/multimedia/gstreamer/common/qgst_p.h#L421) . I /think/ that the nixpkgs derivation takes care of this properly but this is the bundled version. `wrapQt` does not handle this since it is supposed to work for nix-built apps (I think? Otherwise this may be worth adding).